File: client-tutor/src/app/shared/services/location/room.service.ts
import {Injectable} from '@angular/core';
import {AbstractService} from "../abstract.service";
import {HttpClient} from "@angular/common/http";
import {Room} from "../../models/location/room.model";

@Injectable({
  providedIn: 'root'
})
export class RoomService extends AbstractService<Room> {

  override resourceUrl = '/api/rooms';

  constructor(public override httpClient: HttpClient) {
    super(httpClient);
  }
}

File: server/src/main/java/ua/com/alevel/service/location/RoomService.java
package ua.com.alevel.service.location;

import org.springframework.stereotype.Service;
import ua.com.alevel.persistence.crudhelper.CrudRepositoryHelper;
import ua.com.alevel.persistence.entity.location.Room;
import ua.com.alevel.persistence.repository.location.RoomRepository;
import ua.com.alevel.service.AbstractService;

@Service
public class RoomService extends AbstractService<Room, RoomRepository> {

    private final RoomRepository roomRepository;
    private final CrudRepositoryHelper<Room, RoomRepository> crudRepositoryHelper;

    public RoomService(RoomRepository roomRepository, CrudRepositoryHelper<Room, RoomRepository> crudRepositoryHelper) {
        super(roomRepository, crudRepositoryHelper);
        this.roomRepository = roomRepository;
        this.crudRepositoryHelper = crudRepositoryHelper;
    }
}
